packages/core/src/interpreter.ts

```diff
@@ -1011,8 +1013,8 @@ export class Interpreter<
 
     childService
       .onDone((doneEvent) => {
-        this.removeChild(childService.id);
         this.send(toSCXMLEvent(doneEvent as any, { origin: childService.id }));
+        this.removeChild(childService.id);
```

Comment:
any particular reason why this got moved to after the `this.send` call? I'm a little bit afraid that this has a potential of removing from the incorrect state - the one after the `this.send` "settles" and updates `this` with a new state. I could try to think of a failing test case for this tomorrow (not entirely sure if this is an actual possibility yet)

Reply:
Thought it would be safer but I'll move it back for now

Reply:
https://github.com/statelyai/xstate/pull/2508/commits/5025f1599dc1cca86b497481e3a2f4d3710148c8